File: BrainModel.nim
type
  Neuron = object
    id: int
    sensoryInput: int
    encodedData: int

var neurons: array[10, Neuron]

# Encoding process
for i in 0..<neurons.len:
  echo "Enter sensory input data (1 to 10) for neuron ", i+1, " : "
  var inputData: int = readLine(stdin).parseInt
  neurons[i].id = i
  neurons[i].sensoryInput = inputData
  neurons[i].encodedData = inputData

# Neural Plasticity: Each neuron's encoded data is influenced by its neighbours
for i in 0..<neurons.len:
  if i != 0:
    neurons[i].encodedData += neurons[i-1].encodedData
  if i != neurons.high:
    neurons[i].encodedData += neurons[i+1].encodedData

# Retrieval process
for neuron in neurons:
  echo "Retrieved data from neuron ", neuron.id+1, " is ", neuron.encodedData

File: BrainModel.lua
Neuron = {}
Neuron.__index = Neuron

function Neuron.create(id, input)
  local n = {}
  setmetatable(n, Neuron)
  n.id = id
  n.sensoryInput = input
  n.encodedData = input
  return n
end

neurons = {}
for i = 1, 10 do
  io.write("Enter sensory input data (1 to 10) for neuron " .. i .. ": ")
  local inputData = io.read("*n")
  neurons[i] = Neuron.create(i, inputData)
end

-- Neural Plasticity: Each neuron's encoded data is influenced by its neighbours
for i = 2, #neurons do
  neurons[i].encodedData = neurons[i].encodedData + neurons[i - 1].encodedData
end

-- Retrieval process
for _, neuron in ipairs(neurons) do
  print("Retrieved data from neuron " .. neuron.id .. " is " .. neuron.encodedData)
end
